contracts/swap/swap.go

```diff
@@ -76,7 +76,7 @@ func (s *Swap) ValidateCode(ctx context.Context, b bind.ContractBackend, address
 
 // Deploy a Swap contract
 func (s *Swap) Deploy(auth *bind.TransactOpts, backend bind.ContractBackend, owner common.Address) (addr common.Address, tx *types.Transaction, err error) {
-	addr, tx, s.Instance, err = contract.DeploySimpleSwap(auth, backend, owner)
+	addr, tx, s.Instance, err = contract.DeploySimpleSwap(auth, backend, owner, big.NewInt(0))
```

Comment:
I would love to see some comment on what are the constructor arguments and why are you explicitly passing 0

Reply:
In [Pull request #50 ](https://github.com/ethersphere/swap-swear-and-swindle/pull/50/commits/77fe80a02d95dfaa44a6357e5ea163351aa6e967) from Swap-Swear-Swindle, we added this extra parameter. 0 specifies that we use a hardcoded default for the harddeposit timeout duration, which is 1 day. Any value other than 0 will set the default harddeposit timeout duration to the amount of seconds passed in for this chequebook.

Reply:
I will add this as a comment in the code.

Reply:
Why to use a special case for the 1 day timeout duration (which forces to include a check in the constructor of the contract)? It seems to be easier to declare the DEFAULT_HARDDEPPOSIT_TIMEOUT_DURATION = 86400 constant here and pass it to DeploySimpleSwap instead of the 0 value.

Reply:
I think this is rather not intuitive. 

First, why is the parameter only added to the `DeploySimpleSwap` method and not the `Deploy` method? If you defined it as a parameter to the smart contract, then you probably want the caller to set it (can you explain why the caller needs to set it?), and the caller calls `Deploy`, not something implicitly hidden inside the function.

So if the caller is to have the responsibility of setting this parameter right, IMHO she should have all options and clarity on her side.

Second, to me, a value being 0 meaning it has a default of 1 day is rather confusing - because in the go part we don't have that reference. You could just comment it but then if it changes in the SC for some reason the comment needs to change and in fact the go developer might not even know about the change.

. In `go`, there is the notion of `time.Duration`, which exactly defines a duration in time.

Thus my proposal is:
* Add a `hardDepositTimeout time.Duration` parameter to `Deploy` (unless it really is not something the caller needs to set, which I am wondering now - why do we need this? why does the go part need to set it?)
* So if the go code is actually to call this, then we would have a constant `DefaultHardDepositTimeout` in the go code, which is set to `24 * time.Hour` (there is no `time.Day` in go), and use that in the call to `Deploy`
* Inside `Deploy`, as the smart contracts don't know go's `time.Duration` we would have to convert it to `uint`, as this is what the smart contract interface currently requires
* The smart contract would just take the `uint` value when being called.

Open to other argumentation.

Reply:
I changed the .sol file in such a way that 0 does not cause the default to be set. Instead, we set the default as a const in the go sourcecode